data-prepper-plugins/grok-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/grok/GrokProcessor.java

```diff
@@ -97,6 +98,7 @@ public GrokProcessor(final PluginSetting pluginSetting, final ExpressionEvaluato
         this.fieldToGrok = new LinkedHashMap<>();
         this.executorService = executorService;
         this.expressionEvaluator = expressionEvaluator;
+        this.matchFailTag = grokProcessorConfig.getMatchFailTag();
```

Comment:
I would prefer the naming of `tag_on_match_failure`. Also this can be a `List` or `Set`, and we can add all the tags in the Set when there is a failure.

Reply:
I think it's fine to take a list / set